File: lib/norikra/udf/dateformat.rb
require 'java'
require 'norikra/udf'

module Norikra
  module UDF
    class DateFormat < Norikra::UDF::SingleRow
      def self.init
        require 'norikra-udf-dateformat.jar'
      end

      def definition
        ["dateformat", "me.takuti.norikra.udf.DateFormat", "format"]
      end
    end
  end
end

File: java/me/takuti/norikra/udf/DateFormat.java
package me.takuti.norikra.udf;

import java.util.Date;
import java.util.TimeZone;
import java.text.SimpleDateFormat;

public final class DateFormat {
  public static String format(final long timestamp, final String pattern, final String zone) {
    SimpleDateFormat df = new SimpleDateFormat(pattern);
    df.setTimeZone(TimeZone.getTimeZone(zone));

    return df.format(new Date(timestamp));
  }
}
